skrub/_recipe.py

```diff
@@ -0,0 +1,597 @@
+                    log_scale_columns.add(choice_name)
+            all_rows.append(row)
+
+        metadata = {"log_scale_columns": list(log_scale_columns)}
+        all_ordered_param_names = _get_all_param_names(self.get_param_grid())
+        ordered_param_names = [n for n in all_ordered_param_names if n in param_names]
+        table = pd.DataFrame(all_rows, columns=ordered_param_names)
+        result_keys = [
+            "mean_test_score",
+            "std_test_score",
+            "mean_fit_time",
+            "std_fit_time",
+            "mean_score_time",
+            "std_score_time",
+            "mean_train_score",
+            "std_train_score",
+        ]
+        new_names = _join_utils.pick_column_names(table.columns, result_keys)
+        renaming = dict(zip(table.columns, new_names))
+        table.columns = new_names
+        metadata["log_scale_columns"] = [
+            renaming[c] for c in metadata["log_scale_columns"]
+        ]
+        table.insert(0, "mean_test_score", fitted_search.cv_results_["mean_test_score"])
+        if detailed:
+            for k in result_keys[1:]:
+                if k in fitted_search.cv_results_:
+                    table.insert(table.shape[1], k, fitted_search.cv_results_[k])
+        table = table.sort_values("mean_test_score", ascending=False, ignore_index=True)
+        return (table, metadata) if return_metadata else table
+
+    def plot_parallel_coord(
+        self, fitted_search, colorscale=DEFAULT_COLORSCALE, min_score=None
+    ):
+        cv_results, metadata = self.get_cv_results_table(
+            fitted_search, return_metadata=True, detailed=True
+        )
+        cv_results = cv_results.drop(
+            [
+                "std_test_score",
+                "std_fit_time",
+                "std_score_time",
+                "mean_train_score",
+                "std_train_score",
+            ],
+            axis="columns",
+            errors="ignore",
+        )
+        if min_score is not None:
+            cv_results = cv_results[cv_results["mean_test_score"] >= min_score]
+        return plot_parallel_coord(cv_results, metadata, colorscale=colorscale)
+
+    def __repr__(self):
+        n_steps = len(self._steps)
+        predictor_info = ""
+        if self._has_predictor():
+            n_steps -= 1
+            predictor_info = " + predictor"
+        title = (
+            f"{self.__class__.__name__} with {n_steps} transformations{predictor_info}"
+        )
+        underline = "-" * len(title)
+        description = self.get_pipeline_description()
+        return f"{title}\n{underline}\n{description}"
+
+    def add(
+        self,
+        estimator,
+        cols=s.all(),
+        name=None,
+        keep_original=False,
+        rename_columns="{}",
+        allow_reject=False,
+    ):
+        if self._has_predictor():
+            pred_name = self._get_step_names()[-1]
+            raise ValueError(
```

Comment:
Not a high prio: should we allow more flexibility here and have estimators working as transformers? The hard part is making sure that's what the user wants and they are not stacking estimators by mistake.

sklego introduced this concept that might make sense for us: https://github.com/koaning/scikit-lego/blob/main/sklego/meta/estimator_transformer.py